frontend/packages/console-shared/src/utils/pod-utils.ts

```diff
@@ -0,0 +1,47 @@
+import * as _ from 'lodash';
+import {
+  K8sResourceKind,
+  K8sKind,
+  SelfSubjectAccessReviewKind,
+  AccessReviewResourceAttributes,
+} from '@console/internal/module/k8s';
+import { checkAccess } from '@console/internal/components/utils';
+
+export const calculateRadius = (size: number) => {
+  const radius = size / 2; // 60
+  const podStatusStrokeWidth = (8 / 104) * size; //
+  const podStatusInset = (5 / 104) * size;
+  const podStatusOuterRadius = radius - podStatusInset;
+  const podStatusInnerRadius = podStatusOuterRadius - podStatusStrokeWidth;
+  const decoratorRadius = radius * 0.25;
+
+  return {
+    radius,
+    podStatusInnerRadius,
+    podStatusOuterRadius,
+    decoratorRadius,
+  };
+};
+
+export const checkPodEditAccess = (
+  resource: K8sResourceKind,
+  resourceKind: K8sKind,
+  impersonate: string,
+): Promise<SelfSubjectAccessReviewKind> => {
+  if (_.isEmpty(resource) || !resourceKind) {
+    return Promise.resolve(null);
+  }
+  const { name, namespace } = resource.metadata;
+  const resourceAttributes: AccessReviewResourceAttributes = {
+    group: resourceKind.apiGroup,
+    resource: resourceKind.plural,
+    verb: 'patch',
+    name,
+    namespace,
+  };
+  return checkAccess(resourceAttributes, impersonate);
+};
+
+export function createSvgIdUrl(id: string): string {
```

Comment:
This should be a separate utility for svgs.
Won't need this if you can use PF4 tooltips.